ccmlib/common.py

```diff
@@ -395,7 +395,7 @@ def validate_install_dir(install_dir):
         raise ArgumentError('%s does not appear to be a cassandra or dse installation directory' % install_dir)
 
 
-def check_socket_available(itf):
+def check_socket_available(itf, return_on_error=False):
```

Comment:
The "return a bool" interface is much better than the existing one in my opinion. @ptnapoleon you should consider refactoring the uses of this function to make this the default interface.